### ISSUE TITLE
Make it easier to create cloud resources using built-in templates

### ISSUE THREAD

**Issue by Finenyaco** [seal-io/walrus#2149]
**Is your enhancement related to a problem? Please describe.**

Currently, users need to provide a VPC ID to create cloud resources using the built-in templates. Creating and configuring networks can be challenging for novice users.
If the user hasn't provided a VPC ID, can we automatically create a VPC?   Alternatively, can we offer another template to assist operations engineers in quickly creating a VPC?

**Environment**
- Walrus version: main 0a3779654c43dfb1a5d3299c4862502a909f2878
- UI version: dev-c137c63

**Comment by gitlawr** [comment]
With the capability implemented, we should be able to include rules for public cloud services in builtin resource definitions.
Can we support that as well?

**Comment by shanewxy** [comment]
> With the capability implemented, we should be able to include rules for public cloud services in builtin resource definitions. Can we support that as well?

Yes. The infrastructure fields are optional now. Resources required before like vpc would be created by default if not specified. We could add the rules to builtin resource definition after moving templates from walrus-catalog-sandbox to walrus-catalog.

**Comment by gitlawr** [comment]
Great, let's track that in the linked issue

**Comment by Finenyaco** [comment]
**Environment**
- Walrus version: main f1ad5a6dd0a90bd792b6222337a98d37b89ceb7b
- UI version: dev-22fdb32

**Test Result:** Verified in main.

- Resource Group is required in `azure-mysql` template.

<img width="753" alt="image" src="https://github.com/seal-io/walrus/assets/26084756/d3450ddf-1858-4e79-81c8-5c44667bef4c">


- `azure-redis` template needs to update the tag.

<img width="816" alt="image" src="https://github.com/seal-io/walrus/assets/26084756/7375d916-f665-4c04-9a90-9ee63e735863">
